scripts-available/CDB_FederatedServerTables.sql

```diff
@@ -0,0 +1,255 @@
+    
+    --- Grant SELECT to fdw role (TODO: Re-enable if needed)
+    --- EXECUTE FORMAT ('GRANT SELECT ON %I.%I TO %I;', fdw_objects_name, table_name, fdw_objects_name);
+    
+    -- Check id_column is numeric
+    IF NOT @extschema@.__CDB_FS_Column_Is_Integer(src_table, id_column) THEN
+        RAISE EXCEPTION 'non integer id_column "%"', id_column;
+    END IF;
+
+    -- Check if the geom and mercator columns have a geometry type (if provided)
+    IF geom_column IS NOT NULL AND NOT @extschema@.__CDB_FS_Column_Is_Geometry(src_table, geom_column) THEN
+        RAISE EXCEPTION 'non geometry column "%"', geom_column;
+    END IF;
+    IF webmercator_column IS NOT NULL AND NOT @extschema@.__CDB_FS_Column_Is_Geometry(src_table, webmercator_column) THEN
+        RAISE EXCEPTION 'non geometry column "%"', webmercator_column;
+    END IF;
+
+    -- Get a list of columns excluding the id, geom and the_geom_webmercator
+    SELECT ARRAY(
+        SELECT quote_ident(c) FROM @extschema@.__CDB_FS_GetColumns(src_table) AS c
+        WHERE c NOT IN (SELECT * FROM (SELECT unnest(ARRAY[id_column, geom_column, webmercator_column, 'cartodb_id', 'the_geom', 'the_geom_webmercator']) col) carto WHERE carto.col IS NOT NULL)
+    ) INTO rest_of_cols;
+
+    IF geom_column IS NULL
+    THEN
+        geom_expression := 'NULL AS the_geom';
+    ELSIF @postgisschema@.Find_SRID(local_schema::varchar, remote_table::varchar, geom_column::varchar) = 4326
+    THEN
+        geom_expression := format('t.%I AS the_geom', geom_column);
+    ELSE
+        -- It needs an ST_Transform to 4326
+        geom_expression := format('@postgisschema@.ST_Transform(t.%I, 4326) AS the_geom', geom_column);
+    END IF;
+
+    IF webmercator_column IS NULL
+    THEN
+        webmercator_expression := 'NULL AS the_geom_webmercator';
+    ELSIF @postgisschema@.Find_SRID(local_schema::varchar, remote_table::varchar, webmercator_column::varchar) = 3857
+    THEN
+        webmercator_expression := format('t.%I AS the_geom_webmercator', webmercator_column);
+    ELSE
+        -- It needs an ST_Transform to 3857
+        webmercator_expression := format('@postgisschema@.ST_Transform(t.%I, 3857) AS the_geom_webmercator', webmercator_column);
+    END IF;
+
+    -- CARTO columns expressions
+    carto_columns_expression := ARRAY[
+        format('t.%1$I AS cartodb_id', id_column),
+        geom_expression,
+        webmercator_expression
+    ];
+
+    -- Create a view with homogeneous CDB fields
+    BEGIN
+        EXECUTE format(
+            'CREATE OR REPLACE VIEW %1$I AS
+                SELECT %2s
+                FROM %3$s t',
+            local_name,
+            array_to_string(carto_columns_expression || rest_of_cols, ','),
+            src_table
+        );
+    EXCEPTION
+    WHEN insufficient_privilege THEN
+        RAISE EXCEPTION 'Could not import table "%" as "%": "%" already exists', remote_table, local_name, local_name;
```

Comment:
Is "already exists" an accurate error message `WHEN insufficient_privilege`?

Reply:
In this case, yes. I'm not sure why Postgres returns `insufficient_privilege` when creating a view conflicts with an existing table/view, but it does.